File: log/src/main/java/ua/at/tsvetkov/util/Level.java
package ua.at.tsvetkov.util;

/**
 * Created by Alexandr Tsvetkov on 10/22/18.
 */
public enum Level {
    VERBOSE, INFO, DEBUG, WARNING, ERROR, WTF;

    public char getShortName() {
        switch (this) {
            case VERBOSE: {
                return 'V';
            }
            case INFO: {
                return 'I';
            }
            case DEBUG: {
                return 'D';
            }
            case WARNING: {
                return 'W';
            }
            case ERROR: {
                return 'E';
            }
            case WTF: {
                return 'F';
            }
        }
        return '?';
    }
}

File: lib/src/main/java/ua/at/tsvetkov/util/logger/interceptor/Level.kt
package ua.at.tsvetkov.util.logger.interceptor

enum class Level {
    VERBOSE, DEBUG, INFO, WARNING, ERROR, WTF;

    companion object {
        val names: ArrayList<String> = {
            val array = ArrayList<String>()
            values().forEach { array.add(it.name) }
            array
        }.invoke()
    }

}